### ISSUE TITLE
Amazon S3: our s3 lib does not support V4 of the AWS Api

### ISSUE THREAD

**Issue by AlexTr** [unacms/UNA#2064]
m.754325

```
PHP message: PHP Warning: S3::putObject(): [InvalidRequest] The authorization mechanism you have provided is not supported. Please use AWS4-HMAC-SHA256. in /var/www/html/plugins/amazon-s3/S3.php on line 374
```

Try to use the following one:
https://github.com/akeeba/s3

Original Plan – Support for SigV2 ends on June 24, 2019.
Revised Plan – Any new buckets created after June 24, 2020 will not support SigV2 signed requests, although existing buckets will continue to support SigV2 while we work with customers to move off this older request signing method.
https://una.io/page/view-discussion?id=6261

**Comment by gteeling** [comment]
I am completely tech stupid, but does this mean you have resolved the problem and Amazon S3 is now fully functional on Una.io? 

Or does it mean that everyone will have to follow the steps you did in order to utilize Amazon S3 on Una.io?

Thanks

**Comment by AlexTr** [comment]
@gteeling
The error in this message is only happen in some of Amazon datacenters, we tried several US datacenters and they are working just fine and you can use them, however we found it doesn't work correctly in some European datacenters. Other datacenters wasn't tested.

**Comment by AlexTr** [comment]
Fixed in #2952
